src/data_source/notifier.rs

```diff
@@ -160,18 +162,15 @@ pub struct Notifier<T> {
     // concurrent subscriptions go through the high-throughput multi-producer stream implementation,
     // and do not contend for the lock on `FilterSender::subscribers`.
     #[derivative(Debug = "ignore")]
-    pending: UnboundedReceiver<Subscriber<T>>,
-    #[derivative(Debug = "ignore")]
-    subscribe: UnboundedSender<Subscriber<T>>,
```

Comment:
The point of all this is to avoid blocking on an exclusive lock when we subscribe. If we're taking a lock anyways, we can do something much simpler. But I feel like an optimized mpsc channel would still probably be better here.

Reply:
We can get all results but it would have to be up to a certain limit. Is there one you would recommend?

Reply:
Okay - I changed it back to channels